### PR TITLE
Clean up a documentation for the storage blob module.

### PR DESCRIPTION
**Pull request by dhermes**
Many of the links were formatted incorrectly, etc.

**NOTE**: These changes **were** in #3375 but I've moved them here.

### CODE REVIEW

**Comment by dhermes** [comment]
@tseaver @lukesneeringer Can I get a review here?

File: storage/google/cloud/storage/blob.py [hunk 1]
```diff
@@ -839,6 +846,10 @@ def upload_from_file(self, file_obj, rewind=False, size=None,
 
         :raises: :class:`~google.cloud.exceptions.GoogleCloudError`
                  if the upload response returns an error status.
+
+        .. _object versioning: https://cloud.google.com/storage/\
+                               docs/object-versioning
+        .. _lifecycle: https://cloud.google.com/storage/docs/lifecycle
```
Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Would it make sense to put these in the module-scope docstring as well?

File: storage/google/cloud/storage/blob.py [hunk 2]
```diff
@@ -1205,67 +1215,73 @@ def update_storage_class(self, new_class, client=None):
...
-    See: https://tools.ietf.org/html/rfc7234#section-5.2 and
-         https://cloud.google.com/storage/docs/json_api/v1/objects
+    See: `RFC 7234`_ and `API reference docs`_.
 
-    If the property is not set locally, returns ``None``.
+    If the property is not set locally, returns :data:`None`.
 
     :rtype: str or ``NoneType``
+
+    .. _RFC 7234: https://tools.ietf.org/html/rfc7234#section-5.2
```
Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Likewise this one?

File: storage/google/cloud/storage/blob.py [hunk 3]
```diff
@@ -1205,67 +1215,73 @@ def update_storage_class(self, new_class, client=None):
...
-    See: https://tools.ietf.org/html/rfc6266 and
-         https://cloud.google.com/storage/docs/json_api/v1/objects
+    See: `RFC 6266`_ and `API reference docs`_.
 
-    If the property is not set locally, returns ``None``.
+    If the property is not set locally, returns :data:`None`.
 
     :rtype: str or ``NoneType``
+
+    .. _RFC 6266: https://tools.ietf.org/html/rfc7234#section-5.2
```
Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Ditto.

File: storage/google/cloud/storage/blob.py [hunk 4]
```diff
@@ -1205,67 +1215,73 @@ def update_storage_class(self, new_class, client=None):
...
 
-    See: https://tools.ietf.org/html/rfc7231#section-3.1.2.2 and
-         https://cloud.google.com/storage/docs/json_api/v1/objects
+    See: `RFC 7231`_ and `API reference docs`_.
 
     If the property is not set locally, returns ``None``.
 
     :rtype: str or ``NoneType``
+
+    .. _RFC 7231: https://tools.ietf.org/html/rfc7231#section-3.1.2.2
```
Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Ditto.

File: storage/google/cloud/storage/blob.py [hunk 5]
```diff
@@ -1205,67 +1215,73 @@ def update_storage_class(self, new_class, client=None):
...
-    See: https://tools.ietf.org/html/bcp47 and
-         https://cloud.google.com/storage/docs/json_api/v1/objects
+    See: `BCP47`_ and `API reference docs`_.
 
-    If the property is not set locally, returns ``None``.
+    If the property is not set locally, returns :data:`None`.
 
     :rtype: str or ``NoneType``
+
+    .. _BCP47: https://tools.ietf.org/html/bcp47
```
Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Ditto.

File: storage/google/cloud/storage/blob.py [hunk 6]
```diff
@@ -1205,67 +1215,73 @@ def update_storage_class(self, new_class, client=None):
...
-    See: https://tools.ietf.org/html/rfc2616#section-14.17 and
-         https://cloud.google.com/storage/docs/json_api/v1/objects
+    See: `RFC 2616`_ and `API reference docs`_.
 
-    If the property is not set locally, returns ``None``.
+    If the property is not set locally, returns :data:`None`.
 
     :rtype: str or ``NoneType``
+
+    .. _RFC 2616: https://tools.ietf.org/html/rfc2616#section-14.17
```
Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Ditto.

File: storage/google/cloud/storage/blob.py [hunk 7]
```diff
@@ -1205,67 +1215,73 @@ def update_storage_class(self, new_class, client=None):
...
-    See: https://tools.ietf.org/html/rfc4960#appendix-B and
-         https://cloud.google.com/storage/docs/json_api/v1/objects
+    See: `RFC 4960`_ and `API reference docs`_.
 
-    If the property is not set locally, returns ``None``.
+    If the property is not set locally, returns :data:`None`.
 
     :rtype: str or ``NoneType``
+
+    .. _RFC 4960: https://tools.ietf.org/html/rfc4960#appendix-B
```
Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Ditto.

File: storage/google/cloud/storage/blob.py [hunk 8]
```diff
@@ -1287,11 +1303,12 @@ def component_count(self):
...
         """Retrieve the ETag for the object.
 
-        See: https://tools.ietf.org/html/rfc2616#section-3.11 and
-             https://cloud.google.com/storage/docs/json_api/v1/objects
+        See: `RFC 2616 (etags)`_ and `API reference docs`_.
 
         :rtype: str or ``NoneType``
         :returns: The blob etag or ``None`` if the property is not set locally.
+
+        .. _RFC 2616 (etags): https://tools.ietf.org/html/rfc2616#section-3.11
```
Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Ditto

File: storage/google/cloud/storage/blob.py [hunk 9]
```diff
@@ -1324,12 +1341,13 @@ def id(self):
...
 
-    See: https://tools.ietf.org/html/rfc4960#appendix-B and
-         https://cloud.google.com/storage/docs/json_api/v1/objects
+    See: `RFC 1321`_ and `API reference docs`_.
 
     If the property is not set locally, returns ``None``.
 
     :rtype: str or ``NoneType``
+
+    .. _RFC 1321: https://tools.ietf.org/html/rfc1321
```
Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Ditto.

**Comment by dhermes** [comment]
> Would it make sense to put these in the module-scope docstring as well?

@tseaver I only did this for links which are used more than once.

**Pull request merged**